core-bundle/contao/modules/ModuleLogin.php

```diff
@@ -152,11 +153,22 @@ protected function compile()
 		}
 
 		// Only call the authentication utils if there is an active session to prevent starting an empty session
-		if ($request?->hasSession() && ($request->hasPreviousSession() || $request->getSession()->isStarted()))
+		if ($request && $request->hasSession() && ($request->hasPreviousSession() || $request->getSession()->isStarted()))
```

Comment:
This needs to be reverted.

Reply:
This is to improve IDE. My IDE reports invalid access to the next `$request` variable and also inside the IF-condition, because it cannot understand that `$request?->hasSession()` would mean the rest of the access is never done.

Reply:
That sounds like an issue with your IDE then, because mine does not complain about it. 😄 

But in any case, we have agreed not to work around IDE issues in the code, so this definitely has to be reverted.